starlite/cli/main.py

```diff
@@ -0,0 +1,145 @@
+@option("-p", "--port", help="Serve under this port", type=int, default=8000, show_default=True)
+@option("--host", help="Server under this host", default="127.0.0.1", show_default=True)
+@option("--debug", help="Run app in debug mode", is_flag=True)
+def run(
+    reload: bool,
+    port: int,
+    host: str,
+    debug: bool,
+    env: StarliteEnv,
+    app: Starlite,
+) -> None:
+    """Run a Starlite app.
+
+    The app can be either passed as a module path in the form of <module name>.<submodule>:<app instance or factory>,
+    set as an environment variable STARLITE_APP with the same format or automatically discovered from one of these
+    canonical paths: app.py, asgi.py, application.py or app/__init__.py. When autodiscovering application factories,
+    functions with the name `create_app` are considered, or functions that are annotated as returning a `Starlite`
+    instance.
+    """
+
+    try:
+        import uvicorn
+    except ImportError:
+        raise StarliteCLIException("Uvicorn needs to be installed to run an app")  # pylint: disable=W0707
+
+    if debug or env.debug:
+        app.debug = True
+
+    show_app_info(app)
+
+    console.rule("[yellow]Starting server process", align="left")
+
+    uvicorn.run(
+        env.app_path,
+        reload=env.reload or reload,
+        host=env.host or host,
+        port=env.port or port,
+        factory=env.is_app_factory,
+    )
+
+
+@cli.command()
+def routes(app: Starlite) -> None:  # pragma: no cover
+    """Display information about the application's routes."""
+
+    tree = Tree("", hide_root=True)
+
+    for route in sorted(app.routes, key=lambda r: r.path):
+        if isinstance(route, HTTPRoute):
+            branch = tree.add(f"[green]{route.path}[/green] (HTTP)")
+            for handler in route.route_handlers:
+                handler_info = [
+                    f"[blue]{handler.name or handler.handler_name}[/blue]",
+                ]
+
+                if inspect.iscoroutinefunction(unwrap_partial(handler.fn.value)):
+                    handler_info.append("[magenta]async[/magenta]")
+                else:
+                    handler_info.append("[yellow]sync[/yellow]")
+
+                handler_info.append(f'[cyan]{", ".join(sorted(handler.http_methods))}[/cyan]')
+
+                if len(handler.paths) > 1:
+                    for path in handler.paths:
+                        branch.add(" ".join([f"[green]{path}[green]", *handler_info]))
+                else:
+                    branch.add(" ".join(handler_info))
+
+        else:
+            if isinstance(route, WebSocketRoute):
+                route_type = "WS"
+            else:
+                route_type = "ASGI"
+            branch = tree.add(f"[green]{route.path}[/green] ({route_type})")
+            branch.add(f"[blue]{route.route_handler.name or route.route_handler.handler_name}[/blue]")
+
+    console.print(tree)
+
+
+@cli.group()
```

Comment:
Why not split the commands themselves into a commands module? This file is pretty large

Reply:
That's why this is though, isn't it? There's nothing but commands in this module. You could put everything aside from the root `cli` group into another module, but that's not really much of a difference IMO, since that's 10 out of ~200 lines.

Also, 200 lines isn't really that much, is it?

Reply:
But if you use a module you can split the commands into different files. E.g. "session", "route", "openapi" etc. 

Btw, do we have a command to statically generate a schema file?

Reply:
> But if you use a module you can split the commands into different files. E.g. "session", "route", "openapi" etc.

I mean, we could. I just thought it's better to have one 200 lines file than to have 5 with only a couple lines on each, some with only like 10.

If you have a particularly unwieldy command, you can still put it into its own submodule? Just `from .main import cli` and add your stuff there. 

> Btw, do we have a command to statically generate a schema file?

Nope. But sounds like a good subcommand for `openapi` command group!

Reply:
Okay, I have an idea for a more broken down structure. I'll add it in a bit.

Reply:
@Goldziher I updated things. There's now a `commands` submodule, and groups/commands are simply imported in `cli.main` and added to the root `starlite_group` there. This should suffice for all our needs of more complex CLI stuff at the moment.